cmd/patch-release-notify/cmd/root.go

```diff
@@ -0,0 +1,295 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"bytes"
+	"embed"
+	"errors"
+	"fmt"
+	"html/template"
+	"io"
+	"math"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	"k8s.io/release/cmd/schedule-builder/model"
+	"k8s.io/release/pkg/mail"
+	"sigs.k8s.io/release-utils/env"
+	"sigs.k8s.io/release-utils/log"
+	"sigs.k8s.io/yaml"
+)
+
+//go:embed templates/*.tmpl
+var tpls embed.FS
+
+// rootCmd represents the base command when called without any subcommands
+var rootCmd = &cobra.Command{
+	Use:               "patch-release-notify --schedule-path /path/to/schedule.yaml",
+	Short:             "patch-release-notify check the cherry pick deadline and send an email to notify",
+	Example:           "patch-release-notify --schedule-path /path/to/schedule.yaml",
+	SilenceUsage:      true,
+	SilenceErrors:     true,
+	PersistentPreRunE: initLogging,
+	RunE: func(*cobra.Command, []string) error {
+		return run(opts)
+	},
+}
+
+type options struct {
+	sendgridAPIKey string
+	schedulePath   string
+	dayToalert     int
+	name           string
+	email          string
+	nomock         bool
+	logLevel       string
+}
+
+var opts = &options{}
+
+const (
+	sendgridAPIKeyEnvKey = "SENDGRID_API_KEY" //nolint:gosec // this will be provided via env vars
+	layout               = "2006-01-02"
+
+	schedulePathFlag = "schedule-path"
+	nameFlag         = "name"
+	emailFlag        = "email"
+	dayToalertFlag   = "days-to-alert"
+)
+
+var requiredFlags = []string{
+	schedulePathFlag,
```

Comment:
Could we point that to a default remote location without requiring the path to be locally available?